src/NuGet.Clients/NuGet.CommandLine/Commands/HelpCommand.cs

```diff
@@ -144,6 +144,7 @@ public void ViewHelpForCommand(string commandName)
 
                 foreach (KeyValuePair<OptionAttribute, PropertyInfo> o in options)
                 {
+                    if (o.Value.Name == "NoCache") { continue; }
```

Comment:
A more general purpose solution would be to enhance our commandoption class by adding a bool property that indicates to not show this option in the help command, and then just add that check here. Basically something similar to https://github.com/dotnet/sdk/pull/35107/files#diff-15c09ffd510898e37c1ae177803c544ee05ad9bfa2a1b8a47b60284539e3bcd3R151. 

Say add an property bool named `Hidden`.